### PR TITLE
feat: user login log out implementation

### PR DESCRIPTION
**Pull request by neexite**
[AB#60276](https://dev.azure.com/orckestra001/da965986-c85b-43be-9915-219568b000e5/_workitems/edit/60276)

### CODE REVIEW

File: packages/composables/src/useUser/index.ts [hunk 1]
```diff
@@ -20,16 +20,15 @@ const params: UseUserFactoryParams<User, UpdateParams, RegisterParams> = {
     }
 
     if (userToken) {
-      const user = await context.$occ.api.getUser({ userToken });
+      const user = await context.$occ.api.getUser();
```
Comment (InnaBoitsun) [inline packages/composables/src/useUser/index.ts:23]: Please update docs orc-vsf\docs\guide\use-user.md - add some section where discribe the techical approach is used - what cookies is used for Login/Logout functionality/ how they configured, ... and other important information

File: packages/api-client/src/extensions/auth.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                delete req.cookies[AUTH_COOKIE_NAME];
+                return;
+              }
+              const { myAccount } = configuration;
+              // 1 week
+              const expireInSeconds: number = 60 * 60 * 24 * 7;
+              const cookieExpiration: any = Date.now() + (expireInSeconds * 1000);
+
+              const token: string = jwt.sign(
+                tokenData,
```
[the remark this answers is not in the document]
Comment (neexite) [reply packages/api-client/src/extensions/auth.ts:48]: Do we need this data for some reason? jwt handles expiration by self.

File: packages/api-client/src/api/carts/addCoupon/index.ts [hunk 3]
```diff
@@ -1,12 +1,11 @@
...
 // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
 export default async function addCoupon(context, params) {
 
-  const { api, scope, cdnDamProviderConfig, myAccount } = context.config;
-  const { userToken, cartName = 'Default', couponCode } = params;
-  const { id: customerId } = parseUserToken(userToken, myAccount.secretPassphrase);
+  const { api, scope, cdnDamProviderConfig } = context.config;
+  const { cartName = 'Default', couponCode } = params;
+  const { id: customerId } = context.config.auth.getCustomerToken();
   if (!customerId) return null;
```
[the remark this answers is not in the document]
Comment (neexite) [reply packages/api-client/src/api/carts/addCoupon/index.ts:9]: returned a header token-expire: true

File: packages/theme/middleware/is-authenticated.js [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
-    context.redirect('/en-CA');
+  const token = context.$cookies.get('vsf-occ-data');
+
+  try {
+    if (!token || token.isGuest !== false) {
+      context.app.router.push('/');
+      context.redirect('/en-CA');
+    }
+  } catch (ex) {
+    Logger.debug(ex);
```
[the remark this answers is not in the document]
Comment (neexite) [reply packages/theme/middleware/is-authenticated.js:12]: Nope

**Review by napernik — APPROVED** [review]

**Pull request merged**